### PR TITLE
[Tables] Generate code with the latest autorest

### PR DESCRIPTION
**Pull request by YalinLi0312**
- `autorest --python --python-sdks-folder=. C:\git\azure\azure-rest-api-specs\specification\cosmos-db\data-plane\readme.md --models-mode=msrest --azure-validator=False`
- swagger: https://github.com/Azure/azure-rest-api-specs/blob/main/specification/cosmos-db/data-plane/readme.md

### CODE REVIEW

**Review by mccoyp — APPROVED** [review]

**Comment by msyyc** [comment]
@YalinLi0312 Once https://github.com/Azure/azure-rest-api-specs/pull/25770 merged, you don't need special config in cmd line which in more convenient for you

File: sdk/tables/azure-data-tables/azure/data/tables/_table_client.py [hunk 1]
```diff
@@ -318,24 +317,13 @@ def delete_entity(self, *args: Union[TableEntity, str], **kwargs) -> None:
             if row_key is None:
                 row_key = args[1]
 
-        match_condition = kwargs.pop("match_condition", None)
-        etag = kwargs.pop("etag", None)
-        if match_condition and entity and not etag:
-            try:
-                etag = entity.metadata.get("etag", None)
```
Comment (annatisch) [inline sdk/tables/azure-data-tables/azure/data/tables/_table_client.py:325]: We don't want to remove support for this - it would break existing behaviour and also is unrelated to the new code gen API

File: sdk/tables/azure-data-tables/azure/data/tables/_table_client.py [hunk 2]
```diff
@@ -318,24 +317,13 @@ def delete_entity(self, *args: Union[TableEntity, str], **kwargs) -> None:
...
-        )
-
         try:
             self._client.table.delete_entity(
                 table=self.table_name,
                 partition_key=_prepare_key(partition_key),
                 row_key=_prepare_key(row_key),
-                if_match=if_match,
+                etag=kwargs.pop("etag", None),
+                match_condition=kwargs.pop("match_condition", None) or MatchConditions.IfPresent,
```
Comment (annatisch) [inline sdk/tables/azure-data-tables/azure/data/tables/_table_client.py:326]: This current code will be breaking - so we will either need to discuss the addition of `Unconditionally` to the generated handlers, or continue to parse these values ourselves.

File: sdk/tables/azure-data-tables/azure/data/tables/_generated/aio/operations/_operations.py [hunk 3]
```diff
@@ -733,6 +704,12 @@ async def update_entity(  # pylint: disable=inconsistent-return-statements
             409: ResourceExistsError,
             304: ResourceNotModifiedError,
         }
+        if match_condition == MatchConditions.IfNotModified:
+            error_map[412] = ResourceModifiedError
+        elif match_condition == MatchConditions.IfPresent:
+            error_map[412] = ResourceNotFoundError
```
Comment (annatisch) [inline sdk/tables/azure-data-tables/azure/data/tables/_generated/aio/operations/_operations.py:710]: Can we test (or check the recordings) whether calling `update_entity` ever results in a 412 for `ResourceNotFound`?

**Review by annatisch — CHANGES_REQUESTED** [review]

**Comment by YalinLi0312** [comment]
Found an issue in autorest that `etag` get double quoted, the failing tests in ci are because of this reason. So waiting for a release with this autorest fix: https://github.com/Azure/autorest.python/pull/2133

Hi @msyyc, thanks for your help! Do you know the ETA for the autorest release?

File: sdk/tables/azure-data-tables/azure/data/tables/_table_client.py (hunk 1 above)
Comment (YalinLi0312) [reply sdk/tables/azure-data-tables/azure/data/tables/_table_client.py:325]: Added them back

File: sdk/tables/azure-data-tables/azure/data/tables/_table_client.py (hunk 2 above)
Comment (YalinLi0312) [reply sdk/tables/azure-data-tables/azure/data/tables/_table_client.py]: Changed to pass `IfPresent` when "match_condition" is in value `Unconditionally`.

File: sdk/tables/azure-data-tables/azure/data/tables/_generated/aio/operations/_operations.py (hunk 3 above)
Comment (YalinLi0312) [reply sdk/tables/azure-data-tables/azure/data/tables/_generated/aio/operations/_operations.py:710]: We already have the tests for `delete_entity()` and `update_entity()` operations in storage and cosmos, but the status code in response are `404`. 
e.g.
https://github.com/Azure/azure-sdk-for-python/blob/6adefdcac04e52ae3672d21ef4b2392b7d94c3ff/sdk/tables/azure-data-tables/tests/test_table_entity.py#L830-L839
[recording](https://github.com/Azure/azure-sdk-assets/blob/python/tables/azure-data-tables_e1a287b0d2/python/sdk/tables/azure-data-tables/tests/recordings/test_table_entity.pyTestTableEntitytest_update_entity_not_existing.json#L70)

**Comment by msyyc** [comment]
> Found an issue in autorest that `etag` get double quoted, the failing tests in ci are because of this reason. So waiting for a release with this autorest fix: [Azure/autorest.python#2133](https://github.com/Azure/autorest.python/pull/2133)
> 
> Hi @msyyc, thanks for your help! Do you know the ETA for the autorest release?

Please try latest autorest.python 6.7.6: https://www.npmjs.com/package/@autorest/python

File: sdk/tables/azure-data-tables/azure/data/tables/_generated/aio/operations/_operations.py (hunk 3 above)
Comment (annatisch) [reply sdk/tables/azure-data-tables/azure/data/tables/_generated/aio/operations/_operations.py:710]: That test is not the right one - this is the correct test here:
https://github.com/Azure/azure-sdk-for-python/blob/main/sdk/tables/azure-data-tables/tests/test_table_entity.py#L869

Recording:
https://github.com/Azure/azure-sdk-assets/blob/python/tables/azure-data-tables_e1a287b0d2/python/sdk/tables/azure-data-tables/tests/recordings/test_table_entity.pyTestTableEntitytest_update_entity_with_if_doesnt_match.json#L153-L173

Comment (annatisch) [reply sdk/tables/azure-data-tables/azure/data/tables/_generated/aio/operations/_operations.py:710]: @YalinLi0312 - did you take a look at the test above?

Comment (YalinLi0312) [reply sdk/tables/azure-data-tables/azure/data/tables/_generated/aio/operations/_operations.py:710]: I didn't check the result when entity exists, we can get 412 in this case. 

Update: added the same test but with entity does not exist to the same test function.

File: sdk/tables/azure-data-tables/azure/data/tables/_table_batch.py [hunk 4]
```diff
@@ -234,40 +235,37 @@ def update(self, entity: EntityType, mode: Union[str, UpdateMode] = UpdateMode.M
                 :caption: Creating and adding an entity to a Table
         """
         self._verify_partition_key(entity)
-        temp = entity.copy()  # type: ignore
```
Comment (annatisch) [inline sdk/tables/azure-data-tables/azure/data/tables/_table_batch.py:237]: Why was this changed?
The reason this is here is to prevent mutation of the customers dictionary - it may not be necessary, but have you confirm this?

File: sdk/tables/azure-data-tables/azure/data/tables/aio/_table_client_async.py [hunk 5]
```diff
@@ -415,19 +415,18 @@ async def update_entity(
...
-            etag=etag,
-            match_condition=match_condition or MatchConditions.Unconditionally,
+        match_condition = _get_match_condition(
+            etag=etag, match_condition=match_condition or MatchConditions.Unconditionally
         )
-
+        entity = _add_entity_properties(entity)
         partition_key = entity["PartitionKey"]
         row_key = entity["RowKey"]
-        entity = _add_entity_properties(entity)
```
Comment (annatisch) [inline sdk/tables/azure-data-tables/azure/data/tables/aio/_table_client_async.py:430]: This one makes me a bit nervous.... because technically the async client was doing this the right way, and the sync client was doing this incorrectly. I didn't realize we had differences between the clients here.
The `upsert` method is also doing this the other way around. Could we revert this for now and open an issue to explore this as part of the encoder work (or add it to the existing encoder issue)?

File: sdk/tables/azure-data-tables/azure/data/tables/_table_batch.py (hunk 4 above)
Comment (YalinLi0312) [reply sdk/tables/azure-data-tables/azure/data/tables/_table_batch.py:237]: We do "copy" inside _add_entity_properties(), so it's not necessary to "copy" before calling that function.

**Review by annatisch — APPROVED** [review]
